src/src/com/microsoft/aad/adal/AcquireTokenSilentHandler.java

```diff
@@ -0,0 +1,359 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.aad.adal;
+
+import java.io.IOException;
+
+import com.microsoft.aad.adal.AuthenticationRequest.UserIdentifierType;
+
+import android.content.Context;
+
+/**
+ * Internal class handling the detailed acquiretoken silent logic, including cache lookup and also
+ * interact with web request.
+ */
+class AcquireTokenSilentHandler {
+    private static final String TAG = AcquireTokenSilentHandler.class.getSimpleName();
+    
+    private final Context mContext;
+    private final TokenCacheAccessor mTokenCacheAccessor;
+    private final AuthenticationRequest mAuthRequest;
+    
+    private boolean mAttemptedWithFRT = false;
+    private TokenCacheItem mMrrtTokenCacheItem;
+
+    /**
+     * TODO: need to remove. {@link HttpUrlConnectionFactory} provides the possibility to 
```

Comment:
Done. 

---
In reply to: [62908444](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/611#discussion_r62908444) [](ancestors = 62908444)